### ISSUE TITLE
Explore the responsibilites of `net-istio/cmd/webhook` image

### ISSUE THREAD

**Issue by NohaIhab** [canonical/knative-operators#249]
### Context

We need to better understand what knative images are responsible to conclude what integration tests are missing there.

### What needs to get done

Explore and document what `knative-releases/knative.dev/net-istio/cmd/webhook` image is doing / responsible for.

### Definition of Done

Responsibilities of  `knative-releases/knative.dev/net-istio/cmd/webhook` image are well understood and documented.

**Comment by DnPlas** [comment]
## Context

According to the [repository](https://github.com/knative-extensions/net-istio/tree/main?tab=readme-ov-file):

>Knative net-istio defines a KIngress controller for Istio

and `net-istio` is 

> the Istio-based implementation of KIngress.

The KIngress is best described as a [dependency](https://github.com/knative/serving/blob/main/docs/resources-overview.md#dependencies) of Knative Serving:

>Knative Serving depends on an KIngress implementation in order to function. KIngress is pluggable, and implementations exist for a variety of ingress providers, including [Istio](https://github.com/knative-extensions/net-istio), [Contour](https://github.com/knative-extensions/net-contour) and [Kourier](https://github.com/knative-extensions/net-kourier).

Also from the [docs](https://knative.dev/docs/serving/architecture/#networking-layer-and-ingress):

>Knative Serving depends on a Networking Layer that fulfils the [Knative Networking Specification](https://github.com/knative/networking). For this, Knative Serving defines an internal KIngress resource, which acts as an abstraction for different multiple pluggable networking layers.

and

>Ingress in this case, does not refer to the [Kubernetes Ingress Resource](https://kubernetes.io/docs/concepts/services-networking/ingress/). It refers to the concept of exposing external access to a resource on the cluster.

What is [`KIngress`](https://github.com/knative/networking?tab=readme-ov-file#knative-ingress-aka-kingress)?

> The Knative Ingress (KIngress) CRD is based on the Kubernetes Ingress CRD, with the following additions:
    * Traffic splitting
    * Header modification applied to requests based on the traffic split they are assigned to.
    * Host header rewrite
    * Traffic redirection predicates based on a regexp based condition on headers.

## `net-istio-controller Deployment`

> The net-istio-controller deployment reconciles a cluster's ingress into an [Istio virtual service](https://istio.io/docs/reference/config/networking/v1alpha3/virtual-service/).

### Traffic flow

To understand how the `net-istio-controller` works, it is better to refer to the [docs](https://knative.dev/docs/serving/architecture/#traffic-flow-and-dns) and the following image:

![image](https://github.com/user-attachments/assets/23ef73fe-df9b-4f20-a6c5-e4717d9d53ef)

TL;DR: "Each networking layer has a controller that is responsible to watch the KIngress resources and configure the Ingress Gateway accordingly."

## `net-istio-webhook Deployment`

This `Deployment` is just an admission webhook controller that is responsible for [admission control](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/) of the KIngress CR observed by the `net-istio-controller`.

**Comment by DnPlas** [comment]
The information has been reviewed by @orfeas-k, nothing else to do, but feel free to re-open if needed.
